### DISCUSSION TITLE
UVtools and EmguCV usefull extensions share

### DISCUSSION THREAD

**Discussion by sn4k3** [emgucv/emgucv#519] [Show and tell]
My open source project **UVtools** (MSLA/DLP, file analysis, calibration, repair, conversion and manipulation) is an image processing app for 3D resin printers which fixes and process layer images in batch. It can be usefull for someone who is working with same principle as it heavily incorporate EmguCV for almost everything. It can fast process 2000 4K images, while retaining a cache with lower RAM usage, all operations are parallelized.
Project URL: https://github.com/sn4k3/UVtools (Everything related with EmguCV is inside UVtools.Core)

![image](https://user-images.githubusercontent.com/113281/121091681-47702400-c7e2-11eb-8e32-3355ebd075c1.png)

Example of a heatmap generated from a stack of images:
https://github.com/sn4k3/UVtools/blob/master/UVtools.Core/Operations/OperationLayerExportHeatMap.cs#L97
![image](https://user-images.githubusercontent.com/113281/121093705-51475680-c7e5-11eb-9ff1-1f889376ae3d.png)
![image](https://user-images.githubusercontent.com/113281/121094602-d0895a00-c7e6-11eb-9821-d370f7730a3e.png)

Improved PutText function:
![image](https://user-images.githubusercontent.com/113281/121094247-3c1ef780-c7e6-11eb-8af2-5f883f650023.png)

Draw regular polygons with rotation
![image](https://user-images.githubusercontent.com/113281/121249173-34218f00-c89c-11eb-962f-a60180dab960.png)

I have been using EmguCV for a while now and i have made some usefull extensions that i like to share:
https://github.com/sn4k3/UVtools/blob/master/UVtools.Core/Extensions/EmguExtensions.cs

Thanks for this great library!

**Comment by emgucv** [comment]
That's great! Thanks for sharing.

**Reply by sn4k3** [reply]
Just updated the EmguExtensions to support DrawPolygon & DrawRotatedRectangle & DrawRotatedSquare.
I will keep updating, thanks again!
